### PR TITLE
Adds DB connection attributes in spans

### LINKED ISSUE #2270: Add db connection attributes to database span's span data
ref https://github.com/getsentry/team-sdks/issues/19

To get an overview of this work across all SDKs, please visit:
https://github.com/orgs/getsentry/projects/135

For the starfish db module it was determined that we need extra database attributes, specifically about the database connection attributes.

We're going to matching the [otel db conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/database/). Either one of `server.address` or `server.socket.address` is required. If `server.address` is set, then `server.port` should be set. If `server.port` is different than `server.socket.port`, than `server.socket.port` should also be set. In addition, db.name should be set if possible.

| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `server.address` | string | Name of the database host. | `example.com` | Conditionally Required: See alternative attributes below. |
| `server.port` | int | Logical server port number | `80`; `8080`; `443` | Conditionally Required: [1] |
| `server.socket.address` | string | Physical server IP address or Unix socket address. | `10.5.3.2` | See below |
| `server.socket.port` | int | Physical server port. | `16456` | Recommended: If different than `server.port`. |
| `db.name` | string | This attribute is used to report the name of the database being accessed. For commands that switch the database, this should be set to the target database (even if the command fails). | `customers`; `main` | Conditionally Required: If applicable. |

These values should be set on `span.data`.

### PR DESCRIPTION
**Pull request by antonpirker**
Fixes #2270

### CODE REVIEW

File: sentry_sdk/integrations/django/__init__.py [hunk 1]
```diff
@@ -646,8 +646,9 @@ def connect(self):
...
 
 
-# https://github.com/django/django/blob/6a0dc2176f4ebf907e124d433411e52bba39a28e/django/db/backends/base/base.py#L29
-# Avaliable in Django 1.8+
-def _set_db_system_on_span(span, vendor):
-    # type: (Span, str) -> None
+def _set_db_data(span, vendor, connection_params):
+    # type: (Span, str, Dict[str, str]) -> None
     span.set_data(SPANDATA.DB_SYSTEM, vendor)
+    span.set_data(SPANDATA.DB_NAME, connection_params.get("database"))
```
Comment (sentrivana) [inline sentry_sdk/integrations/django/__init__.py:652]: Is the key called `database` for all db backends? Looks like for postgres it's `dbname`: https://github.com/django/django/blob/89c8c2e8311bd31b79c0829055d1e494ab88b801/django/db/backends/postgresql/base.py#L213 Or is this further normalized somewhere?

Comment (sentrivana) [reply sentry_sdk/integrations/django/__init__.py:652]: [MySQL](https://github.com/django/django/blob/89c8c2e8311bd31b79c0829055d1e494ab88b801/django/db/backends/mysql/base.py#L217) seems to use `database`, but [Oracle](https://github.com/django/django/blob/89c8c2e8311bd31b79c0829055d1e494ab88b801/django/db/backends/oracle/base.py#L246C23-L246C23) seems to not surface the dbname in the params at all if I read the function right, just the [OPTIONS](https://docs.djangoproject.com/en/4.2/ref/databases/#id13).

File: sentry_sdk/integrations/django/__init__.py [hunk 2]
```diff
@@ -646,8 +646,9 @@ def connect(self):
...
+    span.set_data(SPANDATA.SERVER_ADDRESS, connection_params.get("host"))
+    span.set_data(SPANDATA.SERVER_PORT, connection_params.get("port"))
```
Comment (sentrivana) [inline sentry_sdk/integrations/django/__init__.py:654]: These might also differ in different backends I think. [MySQL also sometimes sets](https://github.com/django/django/blob/89c8c2e8311bd31b79c0829055d1e494ab88b801/django/db/backends/mysql/base.py#L219C21-L219C32) the `unix_socket`, which we should probably then set on the span as `SPANDATA.SERVER_SOCKET_ADDRESS`.

**Review by sentrivana — CHANGES_REQUESTED** [review]
Looking good! The Django part might need some special casing for the different backends and one or two test cases for Django would also be great.

File: sentry_sdk/integrations/django/__init__.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/django/__init__.py:652]: Good catch. (I tried it with Postgres and it worked, but will check for dbname too)

Comment (antonpirker) [reply sentry_sdk/integrations/django/__init__.py]: Updated my code (also to only set those data items if we have something that is not none.)

other thing, oracle seems to only the the name of the db to find it: https://docs.djangoproject.com/en/4.2/ref/databases/#id13

File: sentry_sdk/integrations/django/__init__.py (hunk 2 above)
Comment (antonpirker) [reply sentry_sdk/integrations/django/__init__.py]: done

**Comment by antonpirker** [comment]
I updated the code to omit data fields that we can not fill. (asked joris about what they like more)

**Comment by antonpirker** [comment]
The chalice deps thing is still not fixed (something has deps to boto3 that are not working with our deps and I have not yet figured out what lib...)

**Comment by antonpirker** [comment]
Also Django tests for new db attributes are still missing.

**Comment by antonpirker** [comment]
Maybe I have some time in the afternoon to have another shot at this, but if not it would be great if you could take over @sentrivana . Sorry for not having everything ready..

**Comment by sentrivana** [comment]
> Maybe I have some time in the afternoon to have another shot at this, but if not it would be great if you could take over @sentrivana . Sorry for not having everything ready..

No worries, I can take over. You should enjoy your vacation. :)

**Pull request merged**